assets/src/edit-story/components/header/title.js

```diff
@@ -58,7 +68,21 @@ function Title() {
       actions: { updateStory },
     }) => ({ title, slug, updateStory })
   );
-  const { setTitleInput } = useHeader();
+  const { setTitleInput, titleInput } = useHeader();
+  const { highlight, onFocusOut } = useHighlights(
+    ({ storyTitle, onFocusOut }) => ({
+      highlight: storyTitle,
+      onFocusOut,
+    })
+  );
+
+  useEffect(() => {
+    if (highlight) {
+      titleInput?.focus();
+    }
+  });
```

Comment:
I think this should re-run when `[highlight]` changes.

Reply:
@zachhale @samwhale When I add a dependency array to the focus checks for checks that have to set the "Design" tab, it will not focus. I figured it was kosher to leave out the dependency array, since it is a visual effect on the DOM which occurs after render and is a cheap logical comparison which does not update the state. 

FYI, The reason it doesn't focus when `highlight` updates is, when switching tabs from the Prepublish tab to the Design tab, the focus is reset due to the tab changes, and because before individual elements are selected, the buttons and inputs on the Design tab aren't available yet. To rectify this and include the dependency array, I would do a compare on the `ref.current`, as well as `highlight`.

I figured this was better than:
```
  const { setTitleInput, titleInput } = useHeader();
  const { highlight, onFocusOut } = useHighlights(
    ({ storyTitle, onFocusOut }) => ({
      highlight: storyTitle,
      onFocusOut,
    })
  );

  useEffect(() => {
    if (highlight) {
      titleInput?.focus();
    }
   }, [highlight, titleInput])
```
...because the comparisons which occur dependency array might be even more comparisons than the single logical check that's inside the `useEffect`. 

I can be convinced otherwise, just thought I'd explain why it was this way before changing it :D

Reply:
^^ So I was trying to get the dependency to work and even adding the `current` to the dependency array has an issue where if you select the same issue twice, the ref to the focusable button/input never changes, so the focus doesn't get updated. So I'm going to leave it unless we can solve for that. I'm moving this focus logic into a hopefully more intuitive hook so we can use this effect more easily.